From dfb046b5d2fed158a42aba34653e62f1dcaa6640 Mon Sep 17 00:00:00 2001
From: Zachary Ware <zach@python.org>
Date: Sat, 12 Sep 2026 03:54:13 +0000
Subject: [PATCH 5/5] gh-156369: Update OpenSSL error ID data (GH-156380)

Only the Modules/_ssl_data_40.h part applies here; the _ssl_data_300.h and _ssl_data_35.h updates were already backported in 396f1ddbb10.

(cherry picked from commit 7cc333439b6426a43cdbca690d9170c9db0d8669)
---
 Modules/_ssl_data_40.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/Modules/_ssl_data_40.h b/Modules/_ssl_data_40.h
index ea615f655bb..6d8631df0a2 100644
--- a/Modules/_ssl_data_40.h
+++ b/Modules/_ssl_data_40.h
@@ -1,6 +1,6 @@
 /* File generated by Tools/ssl/make_ssl_data.py */
-/* Generated on 2026-04-15T09:18:34.696139+00:00 */
-/* Generated from Git commit v3.15.0a8-132-gd14e31ed683 */
+/* Generated on 2026-08-25T19:25:20.590415+00:00 */
+/* Generated from Git commit openssl-4.0.2-0-gf089acdf4 */
 
 /* generated from args.lib2errnum */
 static struct py_ssl_library_code library_codes[] = {
@@ -9163,6 +9163,11 @@ static struct py_ssl_error_code error_codes[] = {
   #else
     {"CRL_ALREADY_DELTA", 11, 127},
   #endif
+  #ifdef X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", ERR_LIB_X509, X509_R_CRL_SIGNATURE_ALGORITHM_MISMATCH},
+  #else
+    {"CRL_SIGNATURE_ALGORITHM_MISMATCH", 11, 147},
+  #endif
   #ifdef X509_R_CRL_VERIFY_FAILURE
     {"CRL_VERIFY_FAILURE", ERR_LIB_X509, X509_R_CRL_VERIFY_FAILURE},
   #else
-- 
2.53.0

